pkg/pod/pod.go

```diff
@@ -98,6 +105,52 @@ func (c *Client) GetPods() (pods []*v1.Pod, err error) {
 	return listPods, nil
 }
 
+// ListPods returns list of all pods
+func (c *Client) ListPods() (pods []*v1.Pod, err error) {
```

Comment:
>ListPods [](start = 17, length = 8)

I am concerned about performance impact. When MIC singleton makes a query that lists pods is one thing. But NMI is running on every node so the impact of such query is multiplied with the number of nodes in the cluster. Is there a way to limit the query to get pods on a specific node?

Reply:
Yes, this is some potential performance issue here. Ideally ListPods should only run once when the nmi pod starts. Maybe at that time, there are not many pods existing. 
If the nmi pod restarts for some reason, it will query all the pods in the cluster, which may cause perf issue. I did some investigation and it seems we cannot query pods by specific node. I will do more investigation on it.

Reply:
So, why not do something about this. BTW: `GetPods()` method above uses `PodWatcher.Lister().List(crdSelector)`
In your code you use `PodWatcher.Informer().GetStore().List()` - why there is a difference about which API used here? I expect those queries to be different except filtering conditions.

---
In reply to: [378592628](https://github.com/Azure/aad-pod-identity/pull/496#discussion_r378592628) [](ancestors = 378592628)